src/components/common/coordsValidate.ts

```diff
@@ -1,5 +1,46 @@
 export function parseCoords(coord): string[] {
   const regExp = /^(\-?\d+(?:\.\d+)?),\s*(\-?\d+(?:\.\d+)?)$/;
   const result = regExp.exec(coord);
-  return result !== null ? result.slice(1) : null;
+  if (result !== null) {
+    return result.slice(1);
+  } else {
+    //it's geopoint/geostring type
+    let geoObject;
+    try {
+      geoObject = JSON.parse(coord);
+    } catch (e) {
+      throw new Error("Invalid GeoJSON object.");
+    }
+    const hasType = geoObject.hasOwnProperty("type");
+    if (hasType) {
+      switch (geoObject.type) {
+        case "Point":
+        case "MultiPoint":
+        case "LineString":
+        case "MultiLineString":
+        case "Polygon":
+        case "MultiPolygon":
+          return geoObject.coordinates;
+        case "FeatureCollection":
+          return geoObject.features;
+        /*
+        default:
+          throw new Error('Invalid GeoJSON object.');
+          */
+      }
+    } else {
+      const hasGeography = geoObject.hasOwnProperty("Geography");
+      if (hasGeography) {
+        return geoObject;
+      } else {
+        const isSDTArray =
```

Comment:
How about `isArray`?

In the context of this library, SDT has no meaning.

Reply:
Done. Addresed in 6aa1ac5